### PR TITLE
Add test for http request to the prometheus api before backup and aft…

### PR DESCRIPTION
**Pull request by venturasr**
**Related issue(s)**
Issue #598

### CODE REVIEW

File: tests/end-to-end/backup-restore-test/backupe2e/prometheus.go [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+
+	if err != nil {
+		log.Fatalf("Unable to get a reponse from the api. \n http response was '%s' (%d) and not OK (200). Error:\n%s\n", resp.Status, resp.StatusCode, err)
+	}
+
+	qresp.UnmarshalJSON(body)
+
+}
+
+func whatIsthisThing(something interface{}) (float64, string, error) {
```
Comment (k15r) [inline tests/end-to-end/backup-restore-test/backupe2e/prometheus.go]: whatIsThisThing

File: tests/end-to-end/backup-restore-test/backupe2e/prometheus.go [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+	if pointInTime != "" {
+		values.Set("time", pointInTime)
+	}
+
+	uri := domain + ":" + port + api
+	client := &http.Client{
+		Timeout: 15 * time.Second,
+	}
+	url := uri + values.Encode()
+	log.Println(url)
```
Comment (k15r) [inline tests/end-to-end/backup-restore-test/backupe2e/prometheus.go]: all lines that you create using log will be shown even when the test does not fail. imho the proper way for such output would be to use the testing object. (which is not yet accessible, so maybe we should somehow pass it to the  end2end tests)

File: tests/end-to-end/backup-restore-test/backupe2e/prometheus.go (hunk 1 above)
Comment (venturasr) [reply tests/end-to-end/backup-restore-test/backupe2e/prometheus.go]: Changed to`whatIsThisThing`

File: tests/end-to-end/backup-restore-test/restore_cluster_backup_test.go [hunk 3]
```diff
@@ -51,9 +51,17 @@ func TestBackupAndRestoreCluster(t *testing.T) {
 
 	for idx, backupTest := range backupTests {
 		testUUID := uuid.New()
+
+		name := string("")
+		if t := reflect.TypeOf(backupTest); t.Kind() == reflect.Ptr {
+			name = "*" + t.Elem().Name()
```
Comment (k15r) [inline tests/end-to-end/backup-restore-test/restore_cluster_backup_test.go]: thats not a valid dns name

regex used for validation is ```'[a-z0-9]([-a-z0-9]*[a-z0-9])?'```

File: tests/end-to-end/backup-restore-test/backupe2e/prometheus.go [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+	}
+
+}
+
+func (pt *prometheusTest) TestResources(namespace string) {
+	qresp := &queryResponse{}
+	qresp.connectToPrometheusApi(pt.domain, pt.port, pt.api, pt.metricQuery, pt.pointInTime.formmattedValue)
+	pt.afterRestore = *qresp
+
+	if len(qresp.Data.Result) > 0 && len(qresp.Data.Result[0].Value) == 2 {
```
[the remark this answers is not in the document]
Comment (venturasr) [reply tests/end-to-end/backup-restore-test/backupe2e/prometheus.go]: go-cmp looks nice, but why, only for comparing two strings. Importing a library only to do that.

**Comment by venturasr** [comment]
Replaced by https://github.com/kyma-project/kyma/pull/3049

**Pull request closed without merging**